mass_editing/models/mass_object.py

```diff
@@ -20,119 +20,116 @@
+        src_obj = self.model_id.model
+        button_name = _('Mass Editing (%s)') % self.name
+        vals['ref_ir_act_window'] = action_obj.create({
+            'name': button_name,
+            'type': 'ir.actions.act_window',
+            'res_model': 'mass.editing.wizard',
+            'src_model': src_obj,
+            'view_type': 'form',
+            'context': "{'mass_editing_object' : %d}" % (self.id),
+            'view_mode': 'form, tree',
+            'target': 'new',
+            'auto_refresh': 1,
+        }).id
+        vals['ref_ir_value'] = self.env['ir.values'].create({
+            'name': button_name,
+            'model': src_obj,
+            'key2': 'client_action_multi',
+            'value': "ir.actions.act_window," + str(vals['ref_ir_act_window']),
+            'object': True,
+        }).id
+        self.write(vals)
         return True
 
-    def unlink_action(self, cr, uid, ids, context=None):
-        for template in self.browse(cr, uid, ids, context=context):
+    @api.multi
+    def unlink_action(self):
+        for mass in self:
             try:
-                if template.ref_ir_act_window:
-                    act_window_obj = self.pool['ir.actions.act_window']
-                    act_window_obj.unlink(
-                        cr, SUPERUSER_ID, [template.ref_ir_act_window.id],
-                        context=context)
-                if template.ref_ir_value:
-                    ir_values_obj = self.pool['ir.values']
-                    ir_values_obj.unlink(
-                        cr, SUPERUSER_ID, template.ref_ir_value.id,
-                        context=context)
+                if mass.ref_ir_act_window:
+                    mass.ref_ir_act_window.unlink()
+                if mass.ref_ir_value:
+                    mass.ref_ir_value.unlink()
             except:
-                raise orm.except_orm(
-                    _("Warning"),
-                    _("Deletion of the action record failed."))
+                raise UserError(_("Deletion of the action record failed."))
         return True
 
-    def unlink(self, cr, uid, ids, context=None):
-        self.unlink_action(cr, uid, ids, context=context)
-        return super(MassObject, self).unlink(cr, uid, ids, context=context)
+    @api.multi
+    def unlink(self):
+        self.unlink_action()
+        return super(MassObject, self).unlink()
 
-    def copy(self, cr, uid, record_id, default=None, context=None):
+    @api.one
+    def copy(self, default):
         if default is None:
             default = {}
-        default.update({'name': '', 'field_ids': []})
-        return super(MassObject, self).copy(
-            cr, uid, record_id, default, context=context)
+        default.update({'name': _("%s (copy)" % self.name), 'field_ids': []})
+        return super(MassObject, self).copy(default)
+
+
+class IrModuleModule(models.Model):
+    _inherit = 'ir.module.module'
+
+    @api.multi
+    def module_uninstall(self):
```

Comment:
Would you mind elaborating on this new method please? Why are the actions not removed with the module during normal uninstall operation?